Portfolio prod

Obinnabii opened the pull request:
### Style the portfolio page to look like a terminal Console and add contents

There are __three main sections__ namely;
- **_about me_**
- **_languages_**
- **_projects_**

**_about me_** and **_languages_** are rendered in _script.js_. They are typed into the _console_ div in _index.html_ using the _console_ typewriter.
**_projects_** is rendered in _index.html_. the individual projects start off invisible. The **addProjects(**_div_name_**)** function makes them visible. it is integrated into the typewriter calls for callback reasons.

Obinnabii committed:
added navigation bar

Obinnabii committed:
gave nave content an appropriate name

Obinnabii committed:
Linked the Typewriter script to the website

Obinnabii committed:
create div for console

Obinnabii committed:
style hyperlinks

Obinnabii committed:
add a hover feature to hyperlinks

Obinnabii committed:
change page background and text color

Obinnabii committed:
added sonme extra fonts

Obinnabii committed:
added comments for fonts

Obinnabii committed:
change page background and text font

Obinnabii committed:
Merge branch 'portfolio_prod' of https://github.com/Obinnabii/my-portfolio into portfolio_prod

Obinnabii committed:
finction to print and function to print header implemented

Obinnabii committed:
function to print header added to hello button

billyeh commented on portfolio/src/main/webapp/index.html:
```diff
@@ -4,19 +4,34 @@
     <div id="content">
+      <header>
+      <!-- Navigation bar at the top -->
+      <nav>
```
Please try to indent these according to the style guide: 2 indents each time we go down a level https://google.github.io/styleguide/htmlcssguide.html#Indentation.

Obinnabii replied:
Sure thing!

Obinnabii replied:
Is there perhaps an auto-styling add on for cloud shell?

billyeh commented on portfolio/src/main/webapp/style.css:
```diff
@@ -1,3 +1,18 @@
+  text-decoration: none;
+}
+
+a:hover{
```
nit: add space before {, here and on line 10.

Obinnabii committed:
setup bootstrap

Obinnabii committed:
removed filler text and formated nav bar

Obinnabii committed:
add style for link hover and nav bar

Obinnabii committed:
changed about me text and simplified JS functions

Obinnabii committed:
link fontawesome icons

Obinnabii committed:
link ocaml icon

Obinnabii committed:
add class for emphasis

Obinnabii committed:
format languages blurb

Obinnabii committed:
project gifs added

Obinnabii committed:
made the project template

Obinnabii committed:
projects div properly created

Obinnabii committed:
create function to show projects

Obinnabii committed:
fix error with projects_blurb divs

Obinnabii committed:
finished second project

Obinnabii committed:
finished final project

Obinnabii committed:
changed project wrapper class name

Obinnabii committed:
add styling for images in the projects div

Obinnabii committed:
change image names

Obinnabii committed:
script for final project

Obinnabii commented:
Added project blurb skeleton here

Obinnabii commented on portfolio/src/main/webapp/script.js:
```diff
@@ -30,12 +30,61 @@ function typify(div_id){
+  //   C/C++ \
+  // </div>\
+
+const projects_blurb = "<div class = 'row'> <p> Below are some projects that I have worked on.</p></div>\
```
added a skeleton for the projects_blurb

Obinnabii commented on portfolio/src/main/webapp/script.js:
```diff
@@ -30,12 +30,61 @@ function typify(div_id){
+      <p class='card-text'>Some quick example text to build on the card title and make up the bulk of the card's content.</p>\
+    </div>\
+  </div>\
+</div>\
```
added a skeleton for the projects_blurb

Obinnabii commented on portfolio/src/main/webapp/index.html:
```diff
@@ -41,6 +41,22 @@
+            </p>
+          </div>
+        </div>
+      </div>
```
format changes in later commits, this setup worked at first but needed to be tweaked

Obinnabii commented on portfolio/src/main/webapp/script.js:
```diff
@@ -73,33 +74,38 @@ const language_blurb = "<div class = 'row'> <p> I have experience with the follo
-</div>\
-"
-
+const projects_blurb = "</div><div class = 'row'> <h2>projects</h2></div><div class = 'row'> <p> Below is a selection of some of my favorite projects.</p>"
```
added headings for the language and projects section

Obinnabii commented on portfolio/src/main/webapp/script.js:
```diff
@@ -30,12 +30,61 @@ function typify(div_id){
+    </div>\
+  </div>"
+
+  // <div class='col col-2'> \
+  //   C/C++ \
+  // </div>\
```
I couldn't find an icon for C/C++. I tried putting it in without an Icon but the text wouldn't show up properly. I wanted it to be vertically aligned in the center so it wouldn't look weird next to all the other languages. I'm still trying to find a solution to this. I'm open to any suggestions

billyeh replied:
Sorry, seeing this comment now, not sure if the suggestion to try another icon was helpful?

Obinnabii replied:
it was!

Obinnabii committed:
Update Project Name

Give the project a more suitable name. it will make the page feel more like you are looking at me and not a project that I worked on.

Obinnabii committed:
Update README.md with portfolio progress.

Obinnabii committed:
Update README.md

fixed issue with header tag

billyeh commented on portfolio/src/main/webapp/index.html:
```diff
@@ -3,26 +3,121 @@
+    <!-- Bootstrap CSS only -->
+    <link rel="stylesheet" href="https://stackpath.bootstrapcdn.com/bootstrap/4.5.0/css/bootstrap.min.css" integrity="sha384-9aIt2nRpC12Uk9gS9baDl411NQApFmC26EwAOH8WgZl5MYYxFfc+NcPb1dKGj7Sk" crossorigin="anonymous">
+    
+    <!-- custom styles -->
```
nit: capitalize Custom

billyeh commented on portfolio/src/main/webapp/index.html:
```diff
@@ -3,26 +3,121 @@
+      <!--projects -->
+      <!--Space invaders-->
+      <div id="project_0" class ="project_wrapper">
+        <div class=" row row-cols-2">
```
nit: remove extra space after "

billyeh commented on portfolio/src/main/webapp/index.html:
```diff
@@ -3,26 +3,121 @@
 
+      <!--projects -->
+      <!--Space invaders-->
+      <div id="project_0" class ="project_wrapper">
```
remove space before =

billyeh commented on portfolio/src/main/webapp/index.html:
```diff
@@ -3,26 +3,121 @@
+        </div>
+      </div>
+
+      <!--stock exchange simulator-->
```
nit: capitalize Stock

billyeh commented on portfolio/src/main/webapp/index.html:
```diff
@@ -3,26 +3,121 @@
+      <!-- Console -->
+      <div id="console"></div>
 
+      <!--projects -->
+      <!--Space invaders-->
```
nit: for all comments, please try to follow the style guide style https://google.github.io/styleguide/htmlcssguide.html and original examples in the file i.e. <!-- (one space after -) (and one space before -) -->

Obinnabii committed:
change CSS to follow style guide

billyeh commented on portfolio/src/main/webapp/index.html:
```diff
@@ -3,26 +3,121 @@
+      </div>
+      
     </div>
+    
+
+
+
+
+
+
+
```
Let's remove the extra whitespace here and below, lines 118-121.

billyeh commented on portfolio/src/main/webapp/script.js:
```diff
@@ -12,17 +12,104 @@
-  // Add it to the page.
-  const greetingContainer = document.getElementById('greeting-container');
-  greetingContainer.innerText = greeting;
+//HELPER_FUNCTION
+/**Create a typewriter from a given div */
```
PTAL at JS style guide on comments https://google.github.io/styleguide/htmlcssguide.html.

Similar to HTML comments, want a space after the // and /*

Here and below.

billyeh commented on portfolio/src/main/webapp/script.js:
```diff
@@ -12,17 +12,104 @@
-  greetingContainer.innerText = greeting;
+//HELPER_FUNCTION
+/**Create a typewriter from a given div */
+function typify(div_id){
```
nit: add space after ) before {

billyeh commented on portfolio/src/main/webapp/script.js:
```diff
@@ -12,17 +12,104 @@
+/**Create a typewriter from a given div */
+function typify(div_id){
+  //get the div
+  var div = document.getElementById(div_id);
```
let or const https://google.github.io/styleguide/jsguide.html#features-use-const-and-let

billyeh commented on portfolio/src/main/webapp/script.js:
```diff
@@ -12,17 +12,104 @@
+const heading_text = "<div class = 'row'> <h1> Obi Abii's Portfolio </h1> </div>"; 
+
+/**About me blurb */
+const about_me_text = "<div class = 'row'> <p>Hi, My name is Obi and this is my <span class='emphasis'>portfolio! <br></span></p>\
+<p>I'm a CS Major at Cornell, currently interning at Google for the Summer of 2020. My main career focus is <span class='emphasis'>software engineering</span>. \
+I am particularly interested in <span class='emphasis'>Machine learning.</span></p></div>";
```
For our HTML in JS, let's try using the template literals in JS? https://google.github.io/styleguide/jsguide.html#features-strings-template-strings. See also https://google.github.io/styleguide/jsguide.html#features-strings-no-line-continuations, no \ continuations.

Below too

billyeh commented on portfolio/src/main/webapp/script.js:
```diff
@@ -12,17 +12,104 @@
+  .pauseFor(300)
+  .typeString(language_blurb) //print language_section
+  .pauseFor(300)
+  .typeString(projects_blurb) //print project header
```
I would lean towards removing these "print ... " comments. They don't really say anything the code doesn't already.

Obinnabii committed:
change CSS to follow style guide

Obinnabii committed:
make HTML Comments follow the style guide

Obinnabii committed:
change spacing in HTML tags to fit style guide

Obinnabii committed:
remove space before '='

Obinnabii committed:
remove space before '='

Obinnabii committed:
format script.js to meet style guide standards

Obinnabii committed:
format script.js to meet style guide standards

Obinnabii committed:
Added C/C++ icon

Obinnabii committed:
Added C/C++ icon

Obinnabii committed:
get rid of commented lines

Obinnabii committed:
get rid of commented lines

Obinnabii committed:
Merge branch 'portfolio_prod' of https://github.com/Obinnabii/my-portfolio into portfolio_prod

billyeh commented on portfolio/src/main/webapp/script.js:
```diff
@@ -12,17 +12,149 @@
+  </div>`
+
+/**
+* blurb containg information about the languages I know 
```
Blurb containing*

billyeh commented on portfolio/src/main/webapp/script.js:
```diff
@@ -12,17 +12,149 @@
+* The id of the div where the above text will be placed 
+* @type {String}
+*/
+const console_id = "console";
```
nit: CONSOLE_ID

billyeh commented on portfolio/src/main/webapp/script.js:
```diff
@@ -12,17 +12,149 @@
+* List of project_ids 
+* @type {!Array<String>}
+*/
+const projects = ['project_0', 'project_1', 'project_2']
```
const PROJECTS

billyeh commented on portfolio/src/main/webapp/style.css:
```diff
@@ -1,9 +1,70 @@
+/* PROJECT SECTION */
+
+.project-wrapper {
+  visibility: hidden;
```
Let's alphabetize the CSS declarations https://google.github.io/styleguide/htmlcssguide.html#Declaration_Order

billyeh reviewed (approved):
It looks really nice now, thanks for addressing the comments!
